lighthouse-viewer/app/src/main.js

```diff
@@ -0,0 +1,505 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* global window, document */
+
+const ReportGenerator = require('../../../lighthouse-core/report/report-generator');
+// TODO: We only need getFilenamePrefix from asset-saver. Tree shake!
+const AssetSaver = require('../../../lighthouse-core/lib/asset-saver.js');
+
+const firebase = require('firebase/app');
+require('firebase/auth');
+const idb = require('idb-keyval');
+
+// Polyfills
+// TODO: dynamically load this based on feature detection.
```

Comment:
dynamically use or dynamically load? Dynamic load may end up a lot of effort for not much win